hips/hip-0012.md

```diff
@@ -1,38 +1,225 @@
+
+The process based on the following ideas:
+
+- Requiring changes to be documented (either Helm 4 proposals, or standard HIPs).
+Both allows Helm communitity members to review changes, and allow Helm maintainers to provide oversight on the functional and architectual implementation of the given change. Ensuring changes are in good standing with respect to this HIP.
+Finally, documented changes should allow contributors to independently execute changes to the agreed proposal in coordination with the release engineer.
+
+- Requiring a timeboxed development effort intends to both promote focus on the most valuable changes that the community/maintainers can deliver within a reasonable timeframe.
+And avoid an "open ended" development cycle that might drag on and/or fizzle out.
+Ideally the majority of changes will be proposed and agree upon early in the development cycle (the planning phase).
+But changes to this initial scope can be ammended during the course of the development lifecycle.
+So long as they can fit within the agreed upon timeframe.
+Making amending the scope an exlicit requirement seeks to ensure that any changes are well thought out.
+For further changes that don't make it into Helm 4, there will be a Helm 5 of course!
+
 
 ## Specification
 
+### Prerequisites
+Before Helm can start the development of Helm v4 there are some things the Helm organization needs to have in order.
+These include:
+
+- Active maintainers who will take the time to review changes.
+Helm has been in an unoffical "maintenance mode" for some time where changes have slowed down and the project needs the people capable of driving a Helm 4.
+- Direction on what is in and out of scope for Helm 4.
+It’s important to be able to know what to say “yes” to and what to say “no” to with reasons for both.
+- A timeline for the development of Helm 4.
+A timeline sets expectations for the time of development.
+If the time for development is open-ended then development could end up in a perl 6 development cycle and just take forever.
+At some point development needs to stop.
+
+This HIP aims to define sucess criteria for these.
+
+### The process / timelines
+
+Helm 4 development will consist of three main phases:
+1. Planning
+1. Feature development
+1. Release preparation
+
+The timeline will be approximately:
+
+1. Marking this HIP as non-draft
+1. Naming of a Release Engineer: Nov. 2024
+1. Kick-off meeting: KubeCon/CloudNativeCon North America 2024 (Salt Lake City)
+1. Earliest that engineering can begin: Nov. 2024
+1. Expected release of Helm v4.0.0 will be planned for the week before KubeCon/CloudNativeCon North America 2025 (expected November 2025)
+
+Prior to the stable release there will be 4-6 weeks of release candidates for the public to test. The time period will be determined by the testing and state of trust in the code leading up to the release candidate window.
```

Comment:
I wonder if this should be synced differently with the release preparation section. That covers betas, RCs, etc. Thoughts?

Reply:
(I thought I responded here previously) I've moved this to the release preparation section